content/en/docs/Reference/file-based-catalogs.md

```diff
@@ -0,0 +1,559 @@
+
+### Extensibility
+
+Another motivation is to enable more extensibility around indexes. The file-based catalog spec is a low-level representation of an index. While it can be maintained directly in its low-level form, we expect many index maintainers to build interesting extensions on top that can be used by their own custom tooling to make all sorts of mutations. For example, one could imagine a tool that translates a high-level API like (mode=semver) down to the low-level file-based catalog format for upgrade edges. Or perhaps an index maintainer needs to customize all of the bundle metadata by adding a new property to bundles that meet a certain criteria.
+
+The OLM developer community will be making use of this extensibility to build more official tooling on top of the low-level APIs, but the major benefit is that index maintainers have this capability as well.
+
+## Specification
+
+### Structure
+
+File-based catalogs can be stored and loaded from directory-based filesystems.
+
+`opm` loads the catalog by walking the root directory and recursing into subdirectories. It attempts to load every file it finds and fails if any errors occur.
+
+Non-catalog files can be ignored using `.indexignore` files, which behave identically to `.gitignore` files. That is, they have the same rules for [patterns](https://git-scm.com/docs/gitignore#_pattern_format) and precedence.
+
+> #### Example `.gitignore` file
+> ```gitignore
+> # Ignore everything except non-object .json and .yaml files
+> **/*
+> !*.json
+> !*.yaml
+> **/objects/*.json
+> **/objects/*.yaml
+> ```
+
+
+Index maintainers have flexibility to chose their desired layout, but the OLM team recommends storing each package's file-based catalog blobs in separate sub-directories. Each individual file can be either JSON or YAML -- it is not necessary for every file in an index to use the same format. 
+
+This layout has the property that each sub-directoriy in the directory hierarchy is a self-contained index, which makes index composition, discovery, and navigation as simple as trivial filesystem operations.
+
+> #### Basic recommended structure
+> ```
+> index
+> ├── pkgA
+> │   └── index.yaml
+> ├── pkgB
+> │   ├── .indexignore
+> │   ├── index.yaml
+> │   └── objects
+> │       └── pkgB.v0.1.0.clusterserviceversion.yaml
+> └── pkgC
+>     └── index.json
+> ```
+
+This `index` could also be trivially included in a parent index by simply copying it into the parent index's root directory.
+
+### Schema
+
+At its core, file-based catalogs use a simple format that can be extended with arbitrary schemas. The format that all file-based catalog blobs must adhere to is the `Meta` schema. The below [cue][cuelang-spec] `_Meta` schema defines all file-based catalog blobs.
```

Comment:
Don't we define these as go structs in the registry repo? Is there any distinct advantage of listing out these schemas as CUE definition rather than just copy pasting the structs we use?

Reply:
CUE is more expressive in terms of what valid values are, but:
1. As currently stated, the CUE specs don't encode _all_ validations, so its not a full spec. (e.g. valid base64 data)
2. CUE is somewhat esoteric still, and few people really know CUE syntax (though I think it is fairly straightforward to grok, at least for what I've written here)

I can make arguments in both directions. I think if I used the Go structs, we might have to add more narrative about what's valid/invalid. I think documenting "just use `opm validate`" isn't enough because that encodes the spec, but it isn't the spec.